### PR TITLE
search --asm: set breakpoint on found instructions

### PR DESCRIPTION
**Pull request by TheLazron**
#2084
`--asmbp` flag to breakpoint found --asm instructions
ss

### CODE REVIEW

File: pwndbg/commands/search.py [hunk 1]
```diff
@@ -299,6 +305,11 @@ def search(
     ):
         if save:
             saved.add(address)
+        if type == "asm" and asmbp:
+            ins = pwndbg.disasm.one(address)
```
Comment (disconnect3d) [inline pwndbg/commands/search.py]: Why do we need to disasm it?

**Review by disconnect3d — COMMENTED** [review]
Can we do it so it can be specified without `--asm`? Just `--asmbp`?

Ideally, it would be nice if it worked with specifying both `--asm --asmbp` and with specifying only `--asmbp` flag.

(So that if someone hits 'arrow up' and just adds `--asmbp` it should work; but if someone just specifies only `--asmbp` it should also work)

File: pwndbg/commands/search.py (hunk 1 above)
Comment (TheLazron) [reply pwndbg/commands/search.py]: oh sorry I overlooked the redundancy while making changes

**Comment by TheLazron** [comment]
> Ideally, it would be nice if it worked with specifying both --asm --asmbp and with specifying only --asmbp flag.

`--asmbp` now can be used as an independent flag to search for asm instructions and then setting a breakpoint and also works with `asm`

changes

File: pwndbg/commands/search.py [hunk 2]
```diff
@@ -299,6 +305,9 @@ def search(
     ):
         if save:
             saved.add(address)
+        if asmbp:
+            # set breakpoint on the instruction
+            gdb.Breakpoint("*%#x" % address, temporary=True)
```
Comment (disconnect3d) [inline pwndbg/commands/search.py]:
```suggestion
            gdb.Breakpoint("*%#x" % address, temporary=False)
```

**Review by disconnect3d — APPROVED** [review]
I think we should set up a standard breakpoint and let user delete it if they do not want it.

I made a change for this. Going to merge when CI passes.

PS: If u can, pls send another PR implementing tests for this :)

**Comment by TheLazron** [comment]
> PS: If u can, pls send another PR implementing tests for this :)

Sure! on it

**Pull request merged**